lib/grizzly/zwave/command_classes/switch_binary.ex

```diff
@@ -6,11 +6,33 @@ defmodule Grizzly.ZWave.CommandClasses.SwitchBinary do
   or off.
   """
 
+  alias Grizzly.ZWave.DecodeError
+
   @behaviour Grizzly.ZWave.CommandClass
 
   @impl Grizzly.ZWave.CommandClass
   def byte(), do: 0x25
 
   @impl Grizzly.ZWave.CommandClass
   def name(), do: :switch_binary
+
+  # Duration encoding
+  #   * 0 -> instantly
+  #   * 1..127 -> seconds
+  #   * 128..254 -> minutes + 127
+  #   * 255 -> factory default (option v2)
+  def duration_to_byte(:default), do: 255
+  def duration_to_byte(seconds) when seconds in 0..127, do: seconds
+
+  def duration_to_byte(seconds) when seconds in 128..7620 do
+    minutes = div(seconds, 60)
+    127 + minutes
+  end
+
```

Comment:
```suggestion
  def duration_to_byte(seconds) when seconds > 7620, do: 254
  def duration_to_byte(seconds) when seconds < 0, do: 0
```

I'm not sure if these values will actually be received, but it seems nice to bound values that might not bounds checked by code upstream from raising.

Reply:
I made the changes and consolidated duration encoding/decoding into a single SwitchSupport module since the Z-Wave specs are shared by SwitchBinary and SwitchMultilevel